### PR TITLE
Make demolition conditional

### LINKED ISSUE #15265: Condition support for demolition
I saw an user the other day trying to duplicate Tanya as an spy and a demolition unit, this would be technicable doable with GrantConditionOnDeploy, but only Infiltrates is conditional.

### PR DESCRIPTION
**Pull request by PunkPun**
Closes #15265

### CODE REVIEW

**Comment by MustaphaTR** [comment]
This should close https://github.com/OpenRA/OpenRA/issues/12604 too, which can already be closed as reverse duplicate of #15265, i guess.

File: OpenRA.Mods.Common/Traits/Demolition.cs [hunk 1]
```diff
@@ -91,15 +93,22 @@ public void ResolveOrder(Actor self, Order order)
 					return;
 			}
 
-			self.QueueActivity(order.Queued, new Demolish(self, order.Target, info.EnterBehaviour, info.DetonationDelay,
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Demolition.cs:94]: Is `order.Queued` save to remove here?

Comment (PunkPun) [reply OpenRA.Mods.Common/Traits/Demolition.cs:94]: Probably not. I didn't notice this overload, reverted

File: OpenRA.Mods.Common/Scripting/Properties/DemolitionProperties.cs [hunk 2]
```diff
@@ -19,21 +19,22 @@ namespace OpenRA.Mods.Common.Scripting
...
 		[ScriptActorPropertyActivity]
 		[Desc("Demolish the target actor.")]
 		public void Demolish(Actor target)
 		{
 			// NB: Scripted actions get no visible targetlines.
-			Self.QueueActivity(new Demolish(Self, Target.FromActor(target), info.EnterBehaviour, info.DetonationDelay,
-				info.Flashes, info.FlashesDelay, info.FlashInterval, info.DamageTypes, null));
+			var demolition = demolitions.FirstEnabledTraitOrDefault();
+			if (demolition != null)
+				Self.QueueActivity(demolition.GetDemolishActivity(Self, Target.FromActor(target)));
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Scripting/Properties/DemolitionProperties.cs]: This breaks the behaviour mentioned in the comment above.

**Review by abcdefg30 — COMMENTED** [review]
Lgtm otherwise.

File: OpenRA.Mods.Common/Scripting/Properties/DemolitionProperties.cs (hunk 2 above)
Comment (PunkPun) [reply OpenRA.Mods.Common/Scripting/Properties/DemolitionProperties.cs]: hmm, I though not calling ShowTargetLines was enough. Fixed

**Review by abcdefg30 — APPROVED** [review]

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/44e820233debaa8e258741db7a9d52592e1b6341...b9a1a36e0c15f8b48b32e7bb9676f6f656983db5)

**Pull request merged**